algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableTrialFactory.java

```diff
@@ -123,16 +135,33 @@ && hasPotentiallyOpenRequirementToInclude(trial.recruitment()) && hasAgeGroupToI
 
     @NotNull
     @VisibleForTesting
-    static Set<String> filterOnRegionsToInclude(@NotNull ClinicalTrial trial, @NotNull Set<CkbRegion> regionsToInclude) {
-        Set<String> countries = Sets.newHashSet();
-        for (Location location : trial.locations()) {
-            if (regionsToInclude.stream().anyMatch(region -> region.includes(location))
-                    && hasPotentiallyOpenRequirementToInclude(location.status())) {
-                countries.add(location.country());
-            }
-        }
+    static Set<Country> filterOnRegionsToInclude(@NotNull ClinicalTrial trial, @NotNull Set<CkbRegion> regionsToInclude,
+            @NotNull CkbFacilityCurationModel ckbFacilityCurationModel) {
+        Map<String, Map<String, Set<String>>> countriesToCitiesToHospitalNames = trial.locations()
+                .stream()
+                .filter(location -> regionsToInclude.stream().anyMatch(region -> region.includes(location))
+                        && hasPotentiallyOpenRequirementToInclude(location.status()))
+                .collect(Collectors.groupingBy(Location::country, Collectors.groupingBy(Location::city, Collectors.mapping(location -> {
+                    if ("Netherlands".equals(location.country())) {
+                        return ckbFacilityCurationModel.curateFacilityName(location);
+                    }
+                    return "";
```

Comment:
I would not make the curation dependent on "Netherlands", I would rather do pass-through in case there is no curation entry. Maybe AUS or USA want to use this curation as well and would be silly if that is not possible due to hard-coding!